js/src/forum/components/DiscussionListPane.js

```diff
@@ -39,20 +40,28 @@ export default class DiscussionListPane extends Component {
     // If the discussion we are viewing is listed in the discussion list, then
     // we will make sure it is visible in the viewport – if it is not we will
     // scroll the list down to it.
-    const $discussion = $list.find('.DiscussionListItem.active');
-    if ($discussion.length) {
-      const listTop = $list.offset().top;
-      const listBottom = listTop + $list.outerHeight();
-      const discussionTop = $discussion.offset().top;
-      const discussionBottom = discussionTop + $discussion.outerHeight();
+    // When coming from another discussion, scroll to the previous postition
+    // to prevent the discussion list jumping around.
+    if (!app.previous.matches(DiscussionPage)) {
+      const $discussion = $list.find('.DiscussionListItem.active');
+      if ($discussion.length) {
+        const listTop = $list.offset().top;
+        const listBottom = listTop + $list.outerHeight();
+        const discussionTop = $discussion.offset().top;
+        const discussionBottom = discussionTop + $discussion.outerHeight();
 
-      if (discussionTop < listTop || discussionBottom > listBottom) {
-        $list.scrollTop($list.scrollTop() - listTop + discussionTop);
+        if (discussionTop < listTop || discussionBottom > listBottom) {
+          $list.scrollTop($list.scrollTop() - listTop + discussionTop);
+        }
       }
+    } else {
+      const top = app.cache.paneScrollTop || 0;
+      $list.scrollTop(top);
     }
   }
 
-  onremove() {
+  onremove(vnode) {
+    app.cache.paneScrollTop = $(vnode.dom).scrollTop();
```

Comment:
Should this be in `onbeforeremove` to be consistent with IndexPage?

Reply:
I did that initially, but `onbeforeremove` gets only called on the Page component, not in the child components. 
see: https://mithril.js.org/lifecycle-methods.html#onbeforeremove